Extensions/Effects/god-ray-pixi-filter.js

```diff
@@ -0,0 +1,15 @@
+gdjs.PixiFiltersTools.registerFilterCreator('Godray', {
+  makePIXIFilter: function() {
+    var center = new PIXI.Point(0, 0);
+    var godray = new PIXI.filters.GodrayFilter({center});
+
+    return godray;
+  },
+  updateParameter: function(filter, parameterName, value) {
+    if (!['light', 'time', 'gain', 'lacunarity', 'parallel', 'angle', 'x', 'y'].includes(parameterName)) return;
+
+    if (['x', 'y'].includes(parameterName)) filter.center[parameterName] = gdjs.PixiFiltersTools.clampValue(value, -1000, 1000);
```

Comment:
For test with < 4 elements, avoid creating an array and do two if checks. In general, avoid in the game engine avoid creating an object/array if not strictly necessary because this creates garbage objects in memory, which then put pressure on the garbage collector which make the game lag on less powerful devices. 

You can either:
- make "dumb" if conditions (let's say 2 to 3 elements).
- create the array as as static member of some variable:

`PIXI.filters.GodrayFilter.supportedParameters = ['light', 'time', 'gain', 'lacunarity', 'parallel', 'angle', 'x', 'y']`

(this also add an indirection in the code, which is why I recommend only if a lot of values).

Reply:
Also that your approach is perfectly valid in the IDE or in a codebase that is handled by webpack or generally that is not a game. But in the game engine, we're optimising for less arrays/objects creation as possible - which makes things sometimes not as pretty as they could be in an app codebase.

Reply:
updated with simple if statements